### PR TITLE
Decrease the memory pressure of completed user tasks

### PR DESCRIPTION
**Pull request by efeg**
Fixes the issue https://github.com/linkedin/cruise-control/issues/389.

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java [hunk 1]
```diff
@@ -164,23 +166,24 @@ public String getJSONString() {
...
+    basicMap.put(AnalyzerUtils.BROKERS, numBrokers());
+    basicMap.put(AnalyzerUtils.REPLICAS, numReplicasInCluster());
+    basicMap.put(AnalyzerUtils.TOPICS, numTopics());
+    List<Statistic> cachedStatistic = Statistic.cachedValues();
     // List of all statistics AVG, MAX, MIN, STD
-    Map<String, Object> allStatMap = new HashMap();
-    for (Statistic stat : Statistic.values()) {
+    Map<String, Object> allStatMap = new HashMap<>(cachedStatistic.size());
+    for (Statistic stat : cachedStatistic) {
       Map<String, Object> resourceMap = new HashMap<>();
```
Comment (kidkun) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java]: we can also initialize the capacity to `cachedStatistic.size()+3`.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/stats/BrokerStats.java [hunk 2]
```diff
@@ -100,12 +93,12 @@ public String getJSONString(CruiseControlParameters parameters) {
   }
 
   @Override
-  public void writeOutputStream(OutputStream out, CruiseControlParameters parameters) {
-    try {
-      out.write(toString().getBytes(StandardCharsets.UTF_8));
-    } catch (IOException e) {
-      LOG.error("Failed to write output stream.", e);
-    }
+  protected void discardIrrelevantAndCacheRelevant(CruiseControlParameters parameters) {
```
[the remark this answers is not in the document]
Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/stats/BrokerStats.java]: We check `parameters.json()` in this method, but maybe you meant that it was unused in `getJSONString(CruiseControlParameters)` (I fixed the latter in this class)?

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java (hunk 1 above)
Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java]: I assume you mean `cachedResources` -- i.e. not `cachedStatistic`.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/stats/BrokerStats.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (kidkun) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/stats/BrokerStats.java]: yes, i mean `getJSONString`, sorry about the confusion.

**Pull request merged**